tests/characters/test_bob_handles_frags.py

```diff
@@ -311,24 +346,296 @@ def test_bob_gathers_and_combines(enacted_federated_policy, federated_bob, feder
+
+
+def test_federated_bob_retrieves_twice_without_retaining_cfrags(federated_bob,
+                                                                federated_alice,
+                                                                capsule_side_channel,
+                                                                enacted_federated_policy,
+                                                                ):
+    # The side channel delivers all that Bob needs at this point:
+    # - A single MessageKit, containing a Capsule
+    # - A representation of the data source
+    capsule_side_channel.reset()
+    the_message_kit = capsule_side_channel()
+
+    alices_verifying_key = federated_alice.stamp.as_umbral_pubkey()
+
+    delivered_cleartexts = federated_bob.retrieve(the_message_kit,
+                                                  enrico=capsule_side_channel.enrico,
                                                   alice_verifying_key=alices_verifying_key,
                                                   label=enacted_federated_policy.label)
 
     # We show that indeed this is the passage originally encrypted by the Enrico.
     assert b"Welcome to flippering number 1." == delivered_cleartexts[0]
+
+    delivered_cleartexts = federated_bob.retrieve(the_message_kit,
+                                                  enrico=capsule_side_channel.enrico,
+                                                  alice_verifying_key=alices_verifying_key,
+                                                  label=enacted_federated_policy.label,
+                                                  use_precedent_work_orders=True)
+
+    # We show that indeed this is the passage originally encrypted by the Enrico.
+    assert b"Welcome to flippering number 1." == delivered_cleartexts[0]
+
+
+def test_federated_bob_retrieves_twice_by_retaining_cfrags(federated_bob,
+                                                           federated_alice,
+                                                           capsule_side_channel,
+                                                           enacted_federated_policy,
+                                                           ):
+    capsule_side_channel.reset()
+    the_message_kit = capsule_side_channel()
+    alices_verifying_key = federated_alice.stamp.as_umbral_pubkey()
+
+    delivered_cleartexts = federated_bob.retrieve(the_message_kit,
+                                                  enrico=capsule_side_channel.enrico,
+                                                  alice_verifying_key=alices_verifying_key,
+                                                  label=enacted_federated_policy.label,
+                                                  retain_cfrags=True)
+    assert b"Welcome to flippering number 1." == delivered_cleartexts[0]
+
+    # Can't retrieve this message again.
+    # Bob needs to either instantiate the message_kit again or use use_attached_cfrags=True.
```

Comment:
So, we're kinda encouraging to not store sensitive materials this way?

Reply:
Right - not without expressly declaring that intent.  

It's not per se a worse way to store sensitive materials than any other way, it just has a different balance of concerns, which we need to clarify in the docs.